packages/kbn-esql-validation-autocomplete/src/shared/esql_to_kibana_type.ts

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+const ESQL_NUMBER_TYPES = [
```

Comment:
How do we know these types? (I know it is irrelevant with the PR). Can we get rid of this casting? (otherwise we might miss the new field types support). Ideally we should not cast to kibana here

Reply:
> we should not cast to kibana here

I agree. That is the ask in https://github.com/elastic/kibana/issues/174710.